### PR TITLE
Local auth mount documentation 

### PR DESCRIPTION
**Pull request by akshya96**

### CODE REVIEW

File: website/content/api-docs/system/auth.mdx [hunk 1]
```diff
@@ -103,8 +103,14 @@ relevant functionality is only supported in Vault Enterprise:
 
 - `local` `(bool: false)` – Specifies if the auth method is local only. Local
   auth methods are not replicated nor (if a secondary) removed by replication.
-  Logins via local auth methods do not make use of identity, i.e. no entity
-  or groups will be attached to the token.
+  Local auth mounts also generate entities for tokens issued. The entity will 
+  be replicated across clusters and the aliases generated on the local auth 
+  mount will be local to the cluster. If the goal of the marking an auth method
```
Comment (taoism4504) [inline website/content/api-docs/system/auth.mdx:108]:
```suggestion
  mount will be local to the cluster. If the goal of marking an auth method 
```

File: website/content/api-docs/system/auth.mdx [hunk 2]
```diff
@@ -103,8 +103,14 @@ relevant functionality is only supported in Vault Enterprise:
 
 - `local` `(bool: false)` – Specifies if the auth method is local only. Local
   auth methods are not replicated nor (if a secondary) removed by replication.
-  Logins via local auth methods do not make use of identity, i.e. no entity
-  or groups will be attached to the token.
+  Local auth mounts also generate entities for tokens issued. The entity will 
+  be replicated across clusters and the aliases generated on the local auth 
+  mount will be local to the cluster. If the goal of the marking an auth method 
+  as `local` was to comply to GDPR guidelines, then care must be taken in not
```
Comment (taoism4504) [inline website/content/api-docs/system/auth.mdx:109]:
```suggestion
  as `local` was to comply with GDPR guidelines, then care must be taken in not 
```

File: website/content/api-docs/system/auth.mdx [hunk 3]
```diff
@@ -103,8 +103,14 @@ relevant functionality is only supported in Vault Enterprise:
 
 - `local` `(bool: false)` – Specifies if the auth method is local only. Local
   auth methods are not replicated nor (if a secondary) removed by replication.
-  Logins via local auth methods do not make use of identity, i.e. no entity
-  or groups will be attached to the token.
+  Local auth mounts also generate entities for tokens issued. The entity will 
+  be replicated across clusters and the aliases generated on the local auth 
+  mount will be local to the cluster. If the goal of the marking an auth method 
+  as `local` was to comply to GDPR guidelines, then care must be taken in not 
+  setting data pertaining to local auth mount or local auth mount aliases in the
```
Comment (taoism4504) [inline website/content/api-docs/system/auth.mdx:110]:
```suggestion
  set the data pertaining to local auth mount or local auth mount aliases in the 
```

File: website/content/api-docs/system/auth.mdx [hunk 4]
```diff
@@ -103,8 +103,14 @@ relevant functionality is only supported in Vault Enterprise:
...
-  Logins via local auth methods do not make use of identity, i.e. no entity
-  or groups will be attached to the token.
+  Local auth mounts also generate entities for tokens issued. The entity will 
+  be replicated across clusters and the aliases generated on the local auth 
+  mount will be local to the cluster. If the goal of the marking an auth method 
+  as `local` was to comply to GDPR guidelines, then care must be taken in not 
+  setting data pertaining to local auth mount or local auth mount aliases in the 
+  metadata of the associated entity. Metadata related to local auth mount aliases 
+  can be stored as `custom_metadata` on the alias itself which will also be 
+  retained local to the cluster.
```
Comment (taoism4504) [inline website/content/api-docs/system/auth.mdx:113]:
```suggestion
  retained locally to the cluster.
```

File: website/content/docs/concepts/identity.mdx [hunk 5]
```diff
@@ -115,9 +115,13 @@ a particular auth mount point.
...
 ## Local Auth Methods
 
-**Vault Enterpris:** If a login is performed using an auth method that was
-mounted locally (using `local=true`), no entities or groups will be associated
-with the resulting token.
+**Vault Enterprise:** All the auth methods will generate an entity by default 
+when a token is being issued, with the exception of token store. This is applicable 
+for both shared mounts and local auth mounts (using `local=true`) when Vault replication 
+is in use.
+If the goal of the marking an auth method as `local` was to comply to GDPR guidelines,
```
Comment (taoism4504) [inline website/content/docs/concepts/identity.mdx:122]:
```suggestion
If the goal of marking an auth method as `local` was to comply to GDPR guidelines, 
```

File: website/content/docs/concepts/identity.mdx [hunk 6]
```diff
@@ -115,9 +115,13 @@ a particular auth mount point.
...
 
-**Vault Enterpris:** If a login is performed using an auth method that was
-mounted locally (using `local=true`), no entities or groups will be associated
-with the resulting token.
+**Vault Enterprise:** All the auth methods will generate an entity by default 
+when a token is being issued, with the exception of token store. This is applicable 
+for both shared mounts and local auth mounts (using `local=true`) when Vault replication 
+is in use.
+If the goal of the marking an auth method as `local` was to comply to GDPR guidelines, 
+then care must be taken in not setting data pertaining to local auth mount or local auth
```
Comment (taoism4504) [inline website/content/docs/concepts/identity.mdx:123]:
```suggestion
then care must be taken to not set the data pertaining to local auth mount or local auth 
```

**Review by taoism4504 — COMMENTED** [review]
A few nits, but otherwise looks good.

**Review by taoism4504 — APPROVED** [review]
Looks good to me

File: website/content/docs/concepts/identity.mdx [hunk 7]
```diff
@@ -115,9 +115,13 @@ a particular auth mount point.
 
 ## Local Auth Methods
 
-**Vault Enterpris:** If a login is performed using an auth method that was
-mounted locally (using `local=true`), no entities or groups will be associated
-with the resulting token.
+**Vault Enterprise:** All the auth methods will generate an entity by default 
+when a token is being issued, with the exception of token store. This is applicable 
+for both shared mounts and local auth mounts (using `local=true`) when Vault replication
```
Comment (vishalnayak) [inline website/content/docs/concepts/identity.mdx:120]:
```suggestion
for both mounts that are shared between clusters and cluster local auth mounts (using `local=true`) when Vault replication 
```

**Review by vishalnayak — APPROVED** [review]

**Pull request merged**